### ISSUE TITLE
use package.json for main file binding instead of using a generated index.js

### ISSUE THREAD

**Issue by jonnysas** [teambit/bit#741]
Now that we are always building a package.json we should consider removing index.js.. any thoughts?

**Comment by jonnysas** [comment]
➤ Gilad Shoham commented:

[Ran Mizrahi](https://app.asana.com/0/483122929446706/483122929446706) We can remove it / only create it if skip-package-json provided. i'm fine with remove or keep it

**Comment by jonnysas** [comment]
➤ Ran Mizrahi commented:

better to only create if skip-package-json is provided.. how much time should it take?

**Comment by jonnysas** [comment]
➤ Gilad Shoham commented:

not much, maybe an hour or two i guess

**Comment by jonnysas** [comment]
➤ David First commented:

It turned out to be more complicated than that. (I worked more than 4 hours on this). Obviously, the implementation is 10 minutes, but fixing all what it broke takes time.
The PR is: https://github.com/teambit/bit/pull/725

I fixed more tests locally, but before I continue any further, I have some questions:

  1. the major issue with my changes is that dists are broken because we don't generate package.json there. Should we generate package.json or leave the index.js files instead?
  2. when an imported component links to a dependency, the content is require('…/dependency-root-dir/index'). With this feature, it won't be valid. So I changed it to point to the main-file. Are you ok with that, or you prefer to point to the root-directory. require('…/dependency-root-dir'). Both will work, the first is quicker (implemented already), the last is more elegant.
  3. what about dependencies that call other dependencies, should we remove the index.js from there as well? currently I didn't remove it, because as I saw in one of the failing tests that there was no package.json.
